### DISCUSSION TITLE
AWS secrets manager

### DISCUSSION THREAD

**Discussion by brackend** [argoproj/argo-cd#10084] [Ideas]
Hi, I've implemented AWS secrets manager for AWS EKS.  It creates a volume with the SM secret.  Is there a way to utilize the mounted secrets in Argocd for the secrets that would otherwise be stored in argocd-secrets.

I could for example create a job that mounts the secret and updates argocd-secrets.  Or another secret and reference the password using $<variable> in configmap.  But would prefer not to be creating additional artifacts outside the existing Helm charts.

It would be great if it were possible to mount the secret directly ithrough Argocd deployment yaml.
```   serviceAccountName: argo-cd-argocd-repo-server
      volumes:
      - name: secrets-store-inline
        csi:
          driver: secrets-store.csi.k8s.io
          readOnly: true
          volumeAttributes:
            secretProviderClass: "test-secrets"

**Comment by shxpark** [comment]
Were you ever able to figure this out?
